tests/unit/metadata/test_table.py

```diff
@@ -36,6 +36,121 @@ def test__init__(self, transformer_mock):
         transformer_mock.assert_any_call(
             dtype=float, rounding=-1, max_value=100, min_value=-50)
 
+    @patch.object(Constraint, 'from_dict')
+    def test__init__sorts_constraints(self, from_dict_mock):
```

Comment:
I would move part of this logic to specific `unit` tests for the exact method instead of proxying the test through the `__init__` call.
At the `__init__` level I would just validate that we are properly calling the methods that do the work, but the internal methods should be tested on their own.